suitcase/hdf5.py

```diff
@@ -96,25 +96,27 @@ def export(headers, filename, stream_name=None, fields=None, timestamps=True, us
                                                 fletcher32=True)
                     data = [e['data'][key] for e in events]
                     data = np.array(data)
-                    try:
-                        # save numerical data
+
+                    if value['dtype'].lower() == 'string':  # 1D of string
+                        data_len = len(data[0])
+                        data = data.astype('|S'+str(data_len))
                         dataset = data_group.create_dataset(
-                            key, data=data, compression='gzip', fletcher32=True)
-                    except TypeError:
-                        try:
-                            # save data with str type, or list of str
-                            if len(data.shape) == 1:
-                                data_len = len(data[0])
-                            else:
-                                data_len = 1
-                                for v in data[0]:
-                                    data_len = max(data_len, len(v))
-                            data = np.array(data).astype('|S'+str(data_len))
+                            key, data=data, compression='gzip')
+                    elif data.dtype.kind in ['S', 'U']:
+                        # 2D of string, we can't tell from dytpe, they are shown as array only.
+                        if data.ndim == 2:
+                            data_len = 1
+                            for v in data[0]:
+                                data_len = max(data_len, len(v))
+                            data = data.astype('|S'+str(data_len))
                             dataset = data_group.create_dataset(
                                 key, data=data, compression='gzip')
-                        except TypeError:
-                            # ignore other types
-                            print('Dataset {} is not saved, as the data type is {}'.format(key, data.dtype))
+                        else:
+                            raise('Array of str with ndim >= 3 can not be saved.')
```

Comment:
Python doesn't let you `raise` without somehow providing a specific type of exception. See for example:

```python
In [1]: raise('asdf')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-be09e45a66d7> in <module>()
----> 1 raise('asdf')

TypeError: exceptions must derive from BaseException
```

Instead do something like:

```python
In [2]: raise ValueError("asdf")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-ccfb515dbf45> in <module>()
----> 1 raise ValueError("asdf")

ValueError: asdf
```